float64 Speed
uint64 Way
